macos/FluentUI/Colors.swift

```diff
@@ -0,0 +1,357 @@
+	@objc public static let textDominant = NSColor(light: gray900, lightHighContrast: .black, dark: .white)
+	/// text color used for titles
+	@objc public static var textPrimary = NSColor(light: gray900, lightHighContrast: .black, dark: gray100, darkHighContrast: .white)
+	/// text color used for subtitles
+	@objc public static let textSecondary = NSColor(light: gray500, lightHighContrast: gray700, dark: gray400, darkHighContrast: gray200)
+	// text color used in disabled state
+	@objc public static let textDisabled = NSColor(light: gray300, lightHighContrast: gray500, dark: gray600, darkHighContrast: gray400)
+	//		/// text appears on top of the surface that uses `Colors.primary` as background color
+	@objc public static let textOnAccent = NSColor(light: .white, dark: .black)
+	
+	/// icon used as call-to-actions without a label attached. They need to reach a minimum contrast ratio 4.5:1 to its background
+	@objc public static let iconPrimary = NSColor(light: gray500, lightHighContrast: gray700, dark: .white)
+	/// icon that are attached to a label and are only used for decorative purposes
+	@objc public static let iconSecondary = NSColor(light: gray400, lightHighContrast: gray600, dark: gray500, darkHighContrast: gray300, darkElevated: gray400)
+	/// icon color used in disabled state
+	@objc public static let iconDisabled = NSColor(light: gray300, lightHighContrast: gray500, dark: gray600, darkHighContrast: gray400)
+	/// icon appears on top of surfaces that uses `Colors.primary` as background color
+	@objc public static let iconOnAccent = NSColor(light: .white, dark: .black)
+	
+	/// In Darkmode, our system use two sets of background colors -- called base and elevated -- to enhance the perception of depath when one dark interface is layered above another.
+	/// The dark base colors are darker, making background interface appear to recede, and the elevate colors are lighter, making foreground interfaces appear to advance
+	
+	@objc public static var surfacePrimary = NSColor(light: .white, dark: .black, darkElevated: gray950)
+	@objc public static let surfaceSecondary = NSColor(light: gray25, dark: gray950, darkElevated: gray900)
+	@objc public static let surfaceTertiary = NSColor(light: gray50, dark: gray900, darkElevated: gray800)
+	/// also used for disabled background color
+	@objc public static let surfaceQuaternary = NSColor(light: gray100, dark: gray600)
+	
+	@objc public static let dividerOnPrimary = NSColor(light: gray100, dark: gray800, darkElevated: gray700)
+	@objc public static let dividerOnSecondary = NSColor(light: gray200, dark: gray700, darkElevated: gray600)
+	@objc public static let dividerOnTertiary = NSColor(light: gray200, dark: gray700, darkElevated: gray600)
+	
+	@objc public static func color(from palette: Palette) -> NSColor {
+		return palette.color
+	}
+	
+	@available(*, unavailable)
+	override init() {
+		super.init()
+	}
```

Comment:
Probably outside of this review, but is there anyway we can avoid this? maybe by making this class a struct or something?